### ISSUE TITLE
Some help avoiding import cycles?

### ISSUE THREAD

**Issue by kleijnweb** [vektra/mockery#201]
I am banging my head on import cycles caused by generated mocks. When an interfaces' signatures reference other types in the mocked interfaces' package, this occurs due to the unit tests being in that same package: `ut (mypackage) > mock (mocks) > inpackagedep (mypackage)`.

How do I get around this once and for all?

**Comment by kleijnweb** [comment]
Use case:

```
package a

type TypeA interface {
  SomeMethod()
}

type TypeB interface {
  SomeMethod() TypeA
}
```

The generated mock will import `a` into `mocks`, making it impossible to also make my UTs in `a`. I've explored these solutions:

1. Using `-inpkg` to generate the mocks in the same pakage.

This works but has some disadvantages:
 - You have some ugly `mock_*` files in your package
 - You cannot use `-all`
 - Messes up code coverage, although "fixable" by deleting the lines from the cover profile afterwards
 - Against some standardization advice I found *out there* about placing all your mocks in a single package.

2. Only mock 1 interface per package.

That probably would make me end up with a separate package for most everything, giving up on cohesion.

3. Not putting my UTs in the same package as UUT

At first I was against this because I wouldn't have access to package privates. I have let that one go (instead viewing package privates in the same light as private class members), but extracting to a higher package doesn't seem like a good fit for UTs (fine for integration tests). So the best thing I got so far is putting my tests in a subpackage of UUTs packages (a/tests > mocks > a). That doesn't seem like how everybody else is doing it though, so I am hesitant to jump on that solution.
